src/declaration.d

```diff
@@ -2049,8 +2056,11 @@ extern (C++) class VarDeclaration : Declaration
 
     final bool isOverlappedWith(VarDeclaration v)
     {
-        return (  offset < v.offset + v.type.size() &&
-                v.offset <   offset +   type.size());
+        const vsz = v.type.size();
+        const tsz = type.size();
+        assert(vsz != SIZE_INVALID && tsz != SIZE_INVALID);
+        return (  offset < v.offset + vsz &&
```

Comment:
Also: `return (expr);` is an unrecommended (we should use a style rule to automatically reject it), use  `return expr;`